File: 백준/Silver/10828. 스택/스택.js
const fs = require("fs");
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
// console.log(input)

let array = []
let result = []

for(let i = 1; i<input.length; i++){
    if(input[i].includes('push')){
        const value = parseInt(input[i].match(/\d+/)?.[0], 10);
        array.push(value)
    }
    else if(input[i] === ('pop')){
        if(array.length === 0){
            result.push(-1)
        }
        else{
            result.push(array.pop())
        }
    }
    else if(input[i] === ('size')){
        result.push(array.length)
    }
    else if(input[i] === ('empty')){
        if(array.length === 0){
            result.push(1)
        }
        else{
            result.push(0)   
        }
    }
    else if(input[i] === ('top')){
        if(array.length === 0){
            result.push(-1)
        }
        else{
            result.push(array[array.length - 1]);
        }
    }
}

console.log(result.join('\n'));

File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

n = int(input())
stack = []
result = [] # 출력용

for i in range(n):
    command = input().split()
    if command[0] == 'push':
        stack.append(int(command[1]))
    elif command[0] == 'pop':
        if len(stack) != 0:
            result.append(str(stack.pop()))
        else:
            result.append('-1')
    elif command[0] == 'size':
        result.append(str(len(stack)))
    elif command[0] == 'empty':
        if len(stack) != 0:
            result.append('0')
        else:
            result.append('1')
    elif command[0] == 'top':
        if len(stack) != 0:
            result.append(str(stack[-1])) # 스택의 가장 위
        else:
            result.append('-1')

print('\n'.join(result))